src/consensus/malachite/read_sync.rs

```diff
@@ -104,6 +104,24 @@ pub struct State {
 
     /// Task for sending status updates
     ticker: JoinHandle<()>,
+
+    initial_sync_completed: bool,
+}
+
+impl State {
+    fn initial_sync_first_completed(&mut self) -> bool {
+        if self.initial_sync_completed {
+            return false;
+        }
+
+        let sync_height = self.sync.sync_height;
+        let peer_ahead_by_threshold = self
+            .sync
+            .random_peer_with_value(sync_height.increment_by(10));
```

Comment:
I don't think we should use a random peer, we should use a validator peer, or we should pick the peer with the highest height (since we know height can never regress)